app/code/Magento/ProductAlert/Block/Email/AbstractEmail.php

```diff
@@ -47,22 +47,37 @@
     protected $imageBuilder;
 
     /**
+     * @var \Magento\Store\Model\App\Emulation
+     */
+    private $appEmulation;
+
+    /**
+     * @var \Magento\Store\Model\StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
      * @param \Magento\Framework\View\Element\Template\Context $context
-     * @param \Magento\Framework\Filter\Input\MaliciousCode $maliciousCode
-     * @param PriceCurrencyInterface $priceCurrency
-     * @param \Magento\Catalog\Block\Product\ImageBuilder $imageBuilder
-     * @param array $data
+     * @param \Magento\Framework\Filter\Input\MaliciousCode    $maliciousCode
+     * @param PriceCurrencyInterface                           $priceCurrency
+     * @param \Magento\Catalog\Block\Product\ImageBuilder      $imageBuilder
+     * @param \Magento\Store\Model\App\Emulation               $appEmulation
+     * @param array                                            $data
      */
     public function __construct(
         \Magento\Framework\View\Element\Template\Context $context,
         \Magento\Framework\Filter\Input\MaliciousCode $maliciousCode,
         PriceCurrencyInterface $priceCurrency,
         \Magento\Catalog\Block\Product\ImageBuilder $imageBuilder,
+        \Magento\Store\Model\App\Emulation $appEmulation,
```

Comment:
* New constructor argument may only be added as the last on the list due to backwards compatibility policy
* Additional argument must be nullable for the same reason.

Please follow [Backwards Compatible Development Guide](http://devdocs.magento.com/guides/v2.1/ext-best-practices/extension-coding/backwards-compatible-development/index.html#adding-a-constructor-parameter) for examples and more information